### ISSUE TITLE
Rule priority

### ISSUE THREAD

**Issue by riley-x** [zokugun/vscode-explicit-folding#104]
Is there a way to define which fold takes precedence when there are multiple matches in one line? For example, looking at the debug output I see

```
[main] line: 273, offset: 0, type: BEGIN, match: ***, regex: 0
[main] line: 273, offset: 12, type: BEGIN, match: ***, regex: 1
[main] line: 277, offset: 0, type: END, match: ***, regex: 1
[main] line: 290, offset: 0, type: END, match: ***, regex: 0
[document] foldings: [{"start":272,"end":276,"kind":3},{"start":272,"end":289,"kind":3}]
```

When I click the fold icon in VS code, it only folds between lines [272, 276]. Instead I would like for it to prioritize the second fold, between [272, 289].

**Comment by daiyam** [comment]
You could use [nested](https://github.com/zokugun/vscode-explicit-folding/blob/master/docs/rules/begin-end.md#nested) but that will depend on your usage.
Can you provide an example (code & rules)?

**Comment by riley-x** [comment]
Silly example:

```js
function t1() { // {{{

}

function t2() {

} // }}}
```

with rules
```json
{ 
      "begin": "{",
      "end": "}",
  },
  {
      "begin": "{{{",
      "end": "}}}",
  },
```

I know here you could just move the location of the comments, but in my use case it doesn't always apply.

**Comment by daiyam** [comment]
Try
```
{
    "begin": "{{{",
    "end": "}}}",
    "nested": [
{ 
    "begin": "{",
    "end": "}",
},
// your other rules
    ]
},
```

**Comment by riley-x** [comment]
This folds the `function t1` line above correctly now, but now `function t2` is not foldable. 

Actually I think just by exchanging the rule order in my above example does what I want:

```json
{
    "begin": "{{{",
    "end": "}}}",
},
{
    "begin": "{",
    "end": "}",
},
```
Is this expected behavior?

**Comment by daiyam** [comment]
> Is this expected behavior?

Yes because if `{` is at the beginning, it will always match the `{{{` 3 times.
Passing `{{{` before `{` will ensure that `{{{` is matched before `{`.

**Comment by daiyam** [comment]
Tell me if it's ok for you and I will merge it...
